app/components/AddRuleModal.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from "react";
+import { Modal } from "./ModalWrapper";
+
+interface AddRuleModalProps {
+  addRequest: (url: string, method: string) => void;
+  addRequestMethod: string;
+  addRequestUrl: string;
+  addRuleErrorNotify: (errorMessage: string, tabId: number) => void;
+  handleClose: () => void;
+  tabId: number;
+  updateAddRequestMethod: (value: string, tabId: number) => void;
+  updateAddRequestUrl: (value: string, tabId: number) => void;
+}
+export const AddRuleModal: React.SFC<AddRuleModalProps> = props => {
```

Comment:
Make it extend from `Component`. 
As already discussed, if we need instance methods never use `Function` component.

Reply:
We are using `functions`(defined internally) and not `methods` or `lifecycle methods` as well.  Do we really need a `stateful` component here ?

Reply:
functions defining in render method is not good, this is similar to it. 
This will define all the methods again and again when each time it renders.

By using `Component` doesn't mean `stateful`. we are not using internal state.